### PR TITLE
netatmo: Refactor netatmo

### PR DESCRIPTION
**Pull request by erikwessman**
I was not able to run pre commit hooks for some reason :+1:

### CODE REVIEW

**Comment by Caodongying** [comment]
> I was not able to run pre commit hooks for some reason 👍

It's because of a missing docstring. I added it so pre-commit is okay now.

File: tests/components/netatmo/test_media_source.py [hunk 1]
```diff
@@ -64,6 +64,9 @@ async def test_async_browse_media(hass: HomeAssistant) -> None:
     # Test successful listing
     media = await async_browse_media(hass, f"{URI_SCHEME}{DOMAIN}/events/")
 
+    # Test successful media children is not None
+    assert media.children
```
Comment (Caodongying) [inline tests/components/netatmo/test_media_source.py:68]: I think media.children can be None if  camera_id is not None and all cameras with the specific event cannot find a media url

**Review by Caodongying — COMMENTED** [review]
I have a question about the new assertion. 🧐

File: tests/components/netatmo/test_media_source.py (hunk 1 above)
Comment (erikwessman) [reply tests/components/netatmo/test_media_source.py:68]: hmmm maybe you are right. but maybe we ignore that 🫣☺️

**Review by eliaskc — APPROVED** [review]

**Pull request merged**